velox/functions/sparksql/fuzzer/CMakeLists.txt

```diff
@@ -12,15 +12,113 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Download Spark source code and copy the Spark connect proto files to Velox
+# directory.
+set(SPARK_VERSION 3.5.1)
+set(TAR_URL
+    "https://github.com/apache/spark/archive/refs/tags/v${SPARK_VERSION}.tar.gz"
+)
+set(TAR_DEST "${CMAKE_BINARY_DIR}/spark-${SPARK_VERSION}.tar.gz")
+
+file(
+  DOWNLOAD ${TAR_URL} ${TAR_DEST}
+  STATUS status
+  LOG log)
+
+list(GET status 0 status_code)
+if(NOT status_code EQUAL 0)
+  message(FATAL_ERROR "Error downloading Spark source file: ${log}.")
+endif()
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E tar xzf ${TAR_DEST}
+                WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
```

Comment:
You can actually do this with FetchContent which will take care of the error codes, unpacking etc. If the archive root has a CMakeLists.txt you can set `SUBDIR` to something that doesn't (iirc something non-existing also works) to avoid building things. You can then use `spark_SOURCE` to copy over the files.

Reply:
Also if we are using a fixed version, should we just vendor the files? I don't have a preference either way. Or is this likely to change (e.g. with every spark release)

Reply:
Thanks for the pointer. Changed to use `FetchContent`.

These files are being updated frequently in Spark, and I assume downloading them from Spark makes it easier for us to update the Spark version later on. cc: @kgpai 
Commit history:
https://github.com/apache/spark/commits/4d13c2231cc4852b922079895223104972dc960b/connector/connect/common/src/main/protobuf/spark/connect